### PR TITLE
fix(android): avoid rebuilds if nothing changed

### PR DESCRIPTION
**Pull request by Flakebi**
Unconditionally overwriting files where the build reruns if they changed leads to rebuilds every time.
Only overwrite a file if its content is different to not rebuild in such a case.

### CODE REVIEW

**Review by lucasfernog — DISMISSED** [review]
nice catch! I did face this issue before but never had the time to actually dig into this, thank you!

File: core/tauri-utils/src/io.rs [hunk 1]
```diff
@@ -4,7 +4,16 @@
 
 //! IO helpers.
 
-use std::io::BufRead;
+use std::{fs::read_to_string, io::BufRead, path::Path};
+
+/// Write to the given file if the content is different.
+pub fn write_if_changed(content: &str, path: &Path) -> std::io::Result<()> {
+  if content != read_to_string(path).unwrap_or_default() {
```
Comment (Flakebi) [inline core/tauri-utils/src/io.rs]: This does a probably unexpected thing if `content == ""`. In that case the file is not created if it doesn’t already exist (due to the `or_default`).
I think the solution from https://github.com/tauri-apps/wry/pull/1342 is better
```rust
if read_to_string(path).map_or(true, |c| c != content) {
  // …
}
```

Comment (lucasfernog) [reply core/tauri-utils/src/io.rs]: right right, let's fix that

Comment (lucasfernog) [reply core/tauri-utils/src/io.rs]: oh actually we already had a function with this functionality, i've pushed a change to use that one instead

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**